### ISSUE TITLE
LaTeX customization

### ISSUE THREAD

**Issue by nschneid** [HappenApps/Quiver#11]
Would it be possible to import additional LaTeX packages, or give the user control over the preamble (as in the [LaTeXiT tool](http://www.chachatelier.fr/latexit/))? This would at minimum allow the user to expand the inventory of available symbols. A user-defined preamble would also allow for custom macros on a per-notebook or application-wide basis.

Thanks for the excellent app!

**Comment by ylian** [comment]
Possible.

Quiver uses MathJax to render LaTeX, and there seems to be an extension for preamble: https://github.com/mathjax/MathJax-third-party-extensions/tree/master/preamble

I am not a LaTeX expert, so it would be helpful if you let me know what you would put in the preamble. I probably need to find out if those extra packages need to be bundled with Quiver.

**Comment by nschneid** [comment]
When writing LaTeX, I tend to add packages on an as-needed basis. Typical ones for symbols include:

- textcomp: may not be necessary with MathJax Unicode support
- amssymb, amsmath: already supported in MathJax
- MnSymbol: e.g., `\llangle` is not currently available
- latexsym: may not be necessary

I sometimes use the `\boldsymbol{}` command, which is listed on the [MathJax page](http://docs.mathjax.org/en/latest/tex.html) but may require a special configuration option.

**Comment by nschneid** [comment]
My other use for the preamble is to define my own macros for use within a document. The MathJax docs mention `\newcommand`; presumably the advantage to a preamble would be to share the same commands across LaTeX cells in a note (or notebook).

**Comment by ylian** [comment]
Adding arbitrary LaTeX packages doesn't seem to be possible with MathJax. http://stackoverflow.com/questions/21192606/can-i-add-latex-packages-to-mathjax

But I have added the extension files so things like `\newcommand` now works. I am adding an application-wide preamble as well. So commands added to this preamble will be available to all LaTeX cells. To share commands within a note, just put those commands in a LaTeX cell inside that note.

These are all implemented in the coming update, Quiver v2.1.

**Comment by nschneid** [comment]
Great, looking forward to it!

**Comment by ylian** [comment]
Implemented in Quiver 2.1. Open preferences and enter custom LaTeX macros.
